modules/ROOT/pages/dataweave-type-selector.adoc

```diff
@@ -0,0 +1,206 @@
+= Type Selector
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Using the dot selector (`.`) over DataWeave xref:dataweave-type-system.adoc#composite-types[types] enables you to declare new types from existing ones. Combined with the xref:dataweave-selectors-reusing-types.adoc[module loader], DataWeave can also load and translate declarations from a custom module file into DataWeave type directives that can be accessed in the same way as types from any other DataWeave module.
+
+== Example: Declaring a New Type from an Existing Type
```

Comment:
Suggest that each example follow the same pattern.

So if we use `== Example: Doing Something` for one heading, we would do that for others. Alternatively, we could eliminate the "Example: " part. Similarly though, if we use something like "== Doing Something" in one place it should be like that everywhere, so for ex, should "Composition," "Union Types," "Generics" also following a similar pattern?